File: backend/src/main/java/com/wedspot/backend/controller/ChatController.java
package com.wedspot.backend.controller;

import com.wedspot.backend.services.IChatService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/v1/chat")
@RequiredArgsConstructor
public class ChatController {

    private final IChatService chatService;

    @GetMapping("/history")
    public ResponseEntity<List<Map<String, Object>>> getChatHistory() {
        return ResponseEntity.ok(chatService.getChatHistory());
    }

    @PostMapping("/message")
    public ResponseEntity<Map<String, Object>> sendMessage(@RequestBody Map<String, String> payload) {
        return ResponseEntity.ok(chatService.sendMessage(payload.get("message")));
    }
}

File: backend/Controllers/ChatController.cs
using Microsoft.AspNetCore.Mvc;
using backend.Services;

namespace backend.Controllers
{
    [ApiController]
    [Route("api/v1/chat")]
    public class ChatController : ControllerBase
    {
        private readonly IChatService _chatService;

        public ChatController(IChatService chatService)
        {
            _chatService = chatService;
        }

        [HttpGet("history")]
        public async Task<ActionResult<List<Dictionary<string, object>>>> GetChatHistory()
        {
            var history = await _chatService.GetChatHistory();
            return Ok(history);
        }

        [HttpPost("message")]
        public async Task<ActionResult<Dictionary<string, object>>> SendMessage([FromBody] Dictionary<string, string> payload)
        {
            if (payload.TryGetValue("message", out var message))
            {
                var response = await _chatService.SendMessage(message);
                return Ok(response);
            }
            return BadRequest("Message is required");
        }
    }
}
